### PR TITLE
Update lunr.js plugin to latest to use different libv8

### PR DESCRIPTION
**Pull request by tyhoff**
This is updating the lunr.js plugin to do two things:

1. Use a libv8 library that works on Mac. This was already done in the
   'master' branch of the plugin, so we just update to the latest
   master.
2. Fix the large amount of JSON nesting by setting max_nesting to false.

---

Error:

```
Installing libv8 3.16.14.19 with native extensions
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    current directory: /Users/tyler/dev/memfault/interrupt/vendor/bundle/ruby/2.7.0/gems/libv8-3.16.14.19/ext/libv8
/Users/tyler/.rbenv/versions/2.7.5/bin/ruby -I /Users/tyler/.rbenv/versions/2.7.5/lib/ruby/2.7.0 -r ./siteconf20230103-55676-1bomqs8.rb
extconf.rb
creating Makefile
/Users/tyler/dev/memfault/interrupt/vendor/bundle/ruby/2.7.0/gems/libv8-3.16.14.19/ext/libv8/builder.rb:86:in `setup_python!': libv8 requires
python 2 to be installed in order to build, but it is currently 3.8.12 (RuntimeError)
	from /Users/tyler/dev/memfault/interrupt/vendor/bundle/ruby/2.7.0/gems/libv8-3.16.14.19/ext/libv8/builder.rb:53:in `build_libv8!'
	from /Users/tyler/dev/memfault/interrupt/vendor/bundle/ruby/2.7.0/gems/libv8-3.16.14.19/ext/libv8/location.rb:24:in `install!'
	from extconf.rb:7:in `<main>'

extconf failed, exit code 1

Gem files will remain installed in /Users/tyler/dev/memfault/interrupt/vendor/bundle/ruby/2.7.0/gems/libv8-3.16.14.19 for inspection.
Results logged to /Users/tyler/dev/memfault/interrupt/vendor/bundle/ruby/2.7.0/extensions/x86_64-darwin-19/2.7.0/libv8-3.16.14.19/gem_make.out

  /Users/tyler/.rbenv/versions/2.7.5/lib/ruby/2.7.0/rubygems/ext/builder.rb:99:in `run'
  /Users/tyler/.rbenv/versions/2.7.5/lib/ruby/2.7.0/rubygems/ext/ext_conf_builder.rb:48:in `block in build'
  /Users/tyler/.rbenv/versions/2.7.5/lib/ruby/2.7.0/tempfile.rb:291:in `open'
  /Users/tyler/.rbenv/versions/2.7.5/lib/ruby/2.7.0/rubygems/ext/ext_conf_builder.rb:30:in `build'
  /Users/tyler/.rbenv/versions/2.7.5/lib/ruby/2.7.0/rubygems/ext/builder.rb:169:in `block in build_extension'
  /Users/tyler/.rbenv/versions/2.7.5/lib/ruby/2.7.0/rubygems/ext/builder.rb:165:in `synchronize'
  /Users/tyler/.rbenv/versions/2.7.5/lib/ruby/2.7.0/rubygems/ext/builder.rb:165:in `build_extension'
  /Users/tyler/.rbenv/versions/2.7.5/lib/ruby/2.7.0/rubygems/ext/builder.rb:210:in `block in build_extensions'
  /Users/tyler/.rbenv/versions/2.7.5/lib/ruby/2.7.0/rubygems/ext/builder.rb:207:in `each'
  /Users/tyler/.rbenv/versions/2.7.5/lib/ruby/2.7.0/rubygems/ext/builder.rb:207:in `build_extensions'
  /Users/tyler/.rbenv/versions/2.7.5/lib/ruby/2.7.0/rubygems/installer.rb:844:in `build_extensions'
  /Users/tyler/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/bundler-2.2.29/lib/bundler/rubygems_gem_installer.rb:66:in `build_extensions'
  /Users/tyler/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/bundler-2.2.29/lib/bundler/rubygems_gem_installer.rb:26:in `install'
  /Users/tyler/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/bundler-2.2.29/lib/bundler/source/rubygems.rb:199:in `install'
  /Users/tyler/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/bundler-2.2.29/lib/bundler/installer/gem_installer.rb:54:in `install'
  /Users/tyler/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/bundler-2.2.29/lib/bundler/installer/gem_installer.rb:16:in `install_from_spec'
  /Users/tyler/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/bundler-2.2.29/lib/bundler/installer/parallel_installer.rb:186:in `do_install'
/Users/tyler/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/bundler-2.2.29/lib/bundler/installer/parallel_installer.rb:177:in `block in
worker_pool'
  /Users/tyler/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/bundler-2.2.29/lib/bundler/worker.rb:62:in `apply_func'
  /Users/tyler/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/bundler-2.2.29/lib/bundler/worker.rb:57:in `block in process_queue'
  /Users/tyler/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/bundler-2.2.29/lib/bundler/worker.rb:54:in `loop'
  /Users/tyler/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/bundler-2.2.29/lib/bundler/worker.rb:54:in `process_queue'
  /Users/tyler/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/bundler-2.2.29/lib/bundler/worker.rb:91:in `block (2 levels) in create_threads'

An error occurred while installing libv8 (3.16.14.19), and Bundler cannot continue.

In Gemfile:
  jekyll-lunr-js-search was resolved to 3.3.0, which depends on
    therubyracer was resolved to 0.12.3, which depends on
      libv8
```

### CODE REVIEW

**Comment by tyhoff** [comment]
Ready for review

**Review by ejohnso49 — COMMENTED** [review]
Might be missing it but I don't see any changes related to `2.`?

File: netlify.toml [hunk 1]
```diff
@@ -6,4 +6,4 @@
 # Deploy Preview context: All Deploy Previews
 # will inherit these settings.
 [context.deploy-preview]
-  command = "jekyll build -D --future && cp _redirects _site/"
+  command = "jekyll build --trace -D --future && cp _redirects _site/"
```
Comment (noahp) [inline netlify.toml:9]: Debug setting?

Comment (tyhoff) [reply netlify.toml:9]: It is a debug setting, but no reason to get rid of it. Will help with future debugging issues if we have them.

**Comment by tyhoff** [comment]
> Might be missing it but I don't see any changes related to `2. Fix the large amount of JSON nesting by setting max_nesting to false.`? Or is that already contained in the our fork of `lunr`?

It's in the fork of lunr. https://github.com/memfault/jekyll-lunr-js-search/commit/c872e3403dd8b885ef513910e4fd03ece7444da8

**Review by noahp — APPROVED** [review]
tyvm! I tested this locally on linux too and it works perfectly :heart:

**Pull request merged**